elastic/kibana
# [Infra] Create typed api client and use it in hosts endpoint

closes https://github.com/elastic/kibana/issues/175717
closes https://github.com/elastic/kibana/issues/175268

 ## Summary

This PR creates a client to wrap searches against metrics indices. With this change, we are able remove most of the manual ES query typing that uses io-ts in favor of the `InferSearchResponseOf` type.

As part of this PR, I refactored the hosts endpoint to use the new client. Other routes will need to be refactored as well.


### How to test

- Start a local Kibana instance
- Navigate to Infrastructure > Hosts
- See if the page loads correctly and sorts by CPU usage desc
